dashboard/src/shared/schema.ts

```diff
@@ -1,17 +1,110 @@
+// WARN: yaml doesn't have updated definitions for TypeScript
+import * as YAML from "yaml";
 import { IBasicFormParam } from "./types";
 
-export function retrieveBasicFormParams(schema: any) {
-  // TBD
-  return {
-    username: {
-      name: "username",
-      path: "wordpressUsername",
-      value: "user",
-    } as IBasicFormParam,
-  };
+// Avoid to explicitly add "null" when an element is not defined
+import { nullOptions } from "yaml/types";
+nullOptions.nullStr = "";
+
+// retrieveBasicFormParams iterates over a JSON Schema properties looking for `form` keys
+// It uses the raw yaml to setup default values.
+// It returns a key:value map for easier handling.
+export function retrieveBasicFormParams(
+  defaultValues: string,
+  schema: any,
+): { [key: string]: IBasicFormParam } {
+  const basicFormParameters = {};
+  if (schema && schema.properties) {
+    const params = lookForFormParams(defaultValues, schema.properties);
+    if (params) {
+      params.forEach(p => (basicFormParameters[p.name] = p));
+    }
+  }
+  return basicFormParameters;
+}
+
+// getDocumentElem returns the YAML element given a root element
+export function getDocumentElem(rootElem: any, path: string): any | undefined {
+  // Split the path per keys
+  const splittedPath = path.split(".");
+  if (splittedPath.length === 1) {
+    // If the path doesn't contain inner objects, simply return
+    return rootElem.get(path);
+  }
+  // If the path contains inner elements, iterate until finding the last one
+  let res = rootElem.get(splittedPath[0]);
+  splittedPath.slice(1, splittedPath.length).forEach(p => (res = res && res.get(p)));
+  return res;
 }
 
+// setValue modifies the current values (text) based on a path
 export function setValue(values: string, path: string, newValue: any) {
```

Comment:
Similarly, https://lodash.com/docs/#set (which also gets around your issue of setting paths which don't exist yet).

It's not clear to my why you need to set the value in a yaml string... why wouldn't you just pass the parsed object (to lodash.set, for example).

Reply:
(replied above)